tools/migrate_branch.py

```diff
@@ -0,0 +1,265 @@
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
+"""
+
+import argparse
+import re
+from . import github_login
+from . import oca_projects
+from .config import read_config
+
+
+class BranchMigrator(object):
+    def __init__(self, source, target, target_org=None, email=None):
+        # Read config
+        config = read_config()
+        self.gh_token = config.get('GitHub', 'token')
+        # Connect to GitHub
+        self.github = github_login.login()
+        gh_user = self.github.user()
+        if not gh_user.email and not email:
+            raise Exception(
+                'Email required to commit to github. Please provide one on '
+                'the command line or make the one of your github profile '
+                'public.')
+        self.gh_credentials = {'name': gh_user.name or str(gh_user),
+                               'email': gh_user.email or email}
+        self.gh_source_branch = source
+        self.gh_target_branch = target
+        self.gh_org = target_org or 'OCA'
+
+    def _replace_content(self, repo, path, replace_list):
+        # Re-read path for retrieving content
+        gh_file = repo.contents(path, self.gh_target_branch)
+        content = gh_file.decoded
+        for replace in replace_list:
+            content = re.sub(replace[0], replace[1], content, flags=re.DOTALL)
+        new_file_blob = repo.create_blob(content, encoding='utf-8')
+        return {
+            'path': path,
+            'mode': '100644',
+            'type': 'blob',
+            'sha': new_file_blob
+        }
+
+    def _create_commit(self, repo, tree_data, message):
+        if tree_data:
+            branch = repo.branch(self.gh_target_branch)
+            tree_sha = branch.commit.commit.tree.sha
+            tree = repo.create_tree(tree_data, tree_sha)
+            commit = repo.create_commit(
+                message=message, tree=tree.sha, parents=[branch.commit.sha],
+                author=self.gh_credentials, committer=self.gh_credentials)
+            repo.ref('heads/{}'.format(branch.name)).update(commit.sha)
+            return commit
+
+    def _mark_modules_uninstallable(self, repo, root_contents):
+        """Make uninstallable the existing modules in the repo."""
+        tree_data = []
+        modules = []
+        for root_content in root_contents.values():
+            if root_content.type != 'dir':
+                continue
+            module_contents = repo.contents(
+                root_content.path, self.gh_target_branch)
+            manifest = module_contents.get('__openerp__.py')
+            if manifest:
+                modules.append(root_content.path)
+                # Re-read path for retrieving content
+                tree_data.append(self._replace_content(
+                    repo, manifest.path,
+                    [('["\"]installable["\"]:[ ]*True',
```

Comment:
Don't you mean '["\']installable["\']:[ ]*True' (with a single quote after the backslash).

What happens if there is no 'installable' key (because True is the default. I always leave it out of my manifests)?